ledger/participant-integration-api/src/main/scala/platform/store/dao/EventProjectionProperties.scala

```diff
@@ -0,0 +1,99 @@
+    populateInterfaceView: Map[String, Map[Identifier, Set[Identifier]]] = Map.empty,
+) {
+  def render(witnesses: Seq[String], templateId: Identifier): RenderResult = {
+    val renderContractArguments = witnesses
+      .map(populateContractArgument.get)
+      .exists {
+        case Some(wildcardTemplates) if wildcardTemplates.isEmpty => true
+        case Some(nonEmptyTemplates) if nonEmptyTemplates(templateId) => true
+        case _ => false
+      }
+
+    val renderInterfaces: Seq[Identifier] =
+      (for {
+        witness <- witnesses.iterator
+        templateToInterfaceMap <- populateInterfaceView.get(witness).toList
+        interfaces <- templateToInterfaceMap.getOrElse(templateId, Set.empty[Identifier])
+      } yield interfaces).toSeq
+
+    RenderResult(renderContractArguments, renderInterfaces)
+  }
+
+}
+
+object EventProjectionProperties {
+
+  case class RenderResult(
+      contractArguments: Boolean,
+      interfaces: Seq[Identifier],
+  )
+
+  def apply(
+      domainTransactionFilter: domain.TransactionFilter,
+      verbose: Boolean,
+      interfaceImplementedBy: Ref.Identifier => Set[Ref.Identifier],
+  ): EventProjectionProperties =
+    EventProjectionProperties(
+      verbose = verbose,
+      populateContractArgument = populateContractArgument(domainTransactionFilter),
+      populateInterfaceView = populateInterfaceView(domainTransactionFilter, interfaceImplementedBy),
+    )
+
+  private def populateContractArgument(domainTransactionFilter: domain.TransactionFilter) = {
+    (for {
+      (party, filters) <- domainTransactionFilter.filtersByParty.iterator
+      inclusiveFilters <- filters.inclusive.iterator
+      templateId <- inclusiveFilters.templateIds.iterator
+    } yield party.toString -> templateId)
+      .toSet[(String, Identifier)]
+      .groupMap(_._1)(_._2)
+      .++(
+        domainTransactionFilter.filtersByParty.iterator
+          .collect {
+            case (party, Filters(None)) =>
+              party.toString -> Set.empty[Identifier]
+
+            case (party, Filters(Some(empty)))
+                if empty.templateIds.isEmpty && empty.interfaceFilters.isEmpty =>
+              party.toString -> Set.empty[Identifier]
+          }
+      )
+  }
+
+  private def populateInterfaceView(
+      domainTransactionFilter: domain.TransactionFilter,
+      interfaceImplementedBy: Identifier => Set[Ref.Identifier],
+  ) = (for {
+    (party, filters) <- domainTransactionFilter.filtersByParty.iterator
+    inclusiveFilters <- filters.inclusive.iterator
+    interfaceFilter <- inclusiveFilters.interfaceFilters.iterator
+    if interfaceFilter.includeView
+    implementor <- interfaceImplementedBy(interfaceFilter.interfaceId).iterator
+  } yield (party, implementor, interfaceFilter.interfaceId))
+    .toSet[(Party, Identifier, Identifier)]
+    .groupMap(_._1) { case (_, templateId, interfaceId) => templateId -> interfaceId }
+    .map { case (key, value) =>
```

Comment:
what do you think about adding here some explanation for readers like `case (partyId, templateAndInterfacePairs: Set[(Identifier, Identifier)] ...`?